src/towncrier/check.py

```diff
@@ -31,12 +31,17 @@ def __main(comparewith, directory, config):
 
     base_directory, config = load_config_from_options(directory, config)
 
+    # handled like this in case the attribute is present but None
+    encoding = getattr(sys.stdout, "encoding", None)
+    if encoding is None:
+        encoding = "utf8"
+
```

Comment:
The comment is confusing...  I use getattr when the attribute might not be present.

Maybe

```suggestion
    encoding = sys.stdout.encoding
    if not encoding:
        # On Python 2.7 or piped standard output.
        encoding = "utf-8"

```

Reply:
Several tests fail in 2.7 if you drop the `getattr()` bit.  So, we need to choose an encoding in that case.  Unless the `.decode()` below shouldn't always happen?  But the extra bit handled here is that if `.encoding` is present but is `None`, we need to pick an encoding.

That said, the comment confused you, and also me for a minute.  So, it needs adjusted.

```suggestion
    # Use UTF-8 both when sys.stdout does not have .encoding (Python 2.7) and
    # when the attribute is present but not None (explicitly piped output and
    # also some CI such as GitHub Actions).
    encoding = getattr(sys.stdout, "encoding", None)
    if encoding is None:
        encoding = "utf8"

```

Reply:
Looks good :)


> # when the attribute is present but not None (explicitly piped output and

# when the attribute is present but **set to** None (explicitly piped output and